frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,194 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Artifact, ArtifactType, Execution } from '@kubeflow/frontend';
+import HelpIcon from '@material-ui/icons/Help';
+import * as React from 'react';
+import { useQuery } from 'react-query';
+import { Link } from 'react-router-dom';
+import IconWithTooltip from 'src/atoms/IconWithTooltip';
+import { color, commonCss, padding } from 'src/Css';
+import { ExecutionHelpers, getOutputArtifactsInExecution } from 'src/lib/MlmdUtils';
+import { filterArtifactsByType, getArtifactTypes } from 'src/lib/OutputArtifactLoader';
+import Banner from '../Banner';
+import { RoutePageFactory } from '../Router';
+import ROCCurve, { DisplayPoint, ROCCurveConfig } from '../viewers/ROCCurve';
+
+const ROC_CURVE_DEFINITION =
+  'The receiver operating characteristic (ROC) curve shows the trade-off between true positive rate and false positive rate. ' +
+  'A lower threshold results in a higher true positive rate (and a higher false positive rate), ' +
+  'while a higher threshold results in a lower true positive rate (and a lower false positive rate)';
+
+type ConfidenceMetrics = {
+  confidenceThreshold: string;
+  falsePositiveRate: number;
+  recall: number;
+};
+
+type MetricsTabProps = {
+  execution: Execution;
+};
+
+interface MetricsSwitcherProps {
+  artifacts: Artifact[];
+  artifactTypes: ArtifactType[];
+}
+
+/**
+ * Metrics tab renders metrics for the artifact of given execution.
+ * Some system metrics are: Confusion Matrix, ROC Curve, Scalar, etc.
+ * Detail can be found in https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/io_types.py
+ * Note that these metrics are only available on KFP v2 mode.
+ */
+export function MetricsTab({ execution }: MetricsTabProps) {
+  // Retriving a list of artifacts associated with this execution,
+  // so we can find the artifact for system metrics from there.
+  const { isLoading: isLoadingArtifacts, error: errorArtifacts, data: artifacts } = useQuery<
+    Artifact[],
+    Error
+  >(['execution_output_artifact', { id: execution.getId() }], () =>
+    getOutputArtifactsInExecution(execution),
+  );
+
+  // artifactTypes allows us to map from artifactIds to artifactTypeNames,
+  // so we can identify metrics artifact provided by system.
+  const {
+    isLoading: isLoadingArtifactTypes,
+    error: errorArtifactTypes,
+    data: artifactTypes,
+  } = useQuery<ArtifactType[], Error>(['artifact_types'], () => getArtifactTypes());
```

Comment:
note, the entire app uses the same cache for fetching artifact types, but artifact types may change over time.

For example, you load the UI first before any v2 pipelines are run, so when you get artifact_types, none of the system types exist. Then you run some pipelines, and look at the visualization, but 'artifact_types' is still cached.

You might want to have a look at https://react-query.tanstack.com/reference/useQuery or other documentation to see what to do with this, probably disable this cache or set a stale time or invalidate cache when we know data is outdated. It's up-to-you to decide what to do now, we can always improve later too.

Reply:
That is a good point, how about I show refresh button on the banner when there is no available metrics? So users can manually clear the cache based on their need.

Reply:
I found this tool useful https://react-query.tanstack.com/devtools to analyze react query staleness.

Reply:
Discussed offline, we are going to use `staleTime infinity` and `enabled` options based on our usecase.